api-put-object-streaming.go

```diff
@@ -49,7 +49,7 @@ func (c Client) putObjectMultipartStream(bucketName, objectName string,
 	// while it is going to be duck typed similar to io.ReaderAt.
 	// It is to indicate that *minio.Object implements io.ReaderAt.
 	// and such a functionality is used in the subsequent code path.
-	if isObject(reader) || isReadAt(reader) || isFile(reader) {
+	if isFile(reader) || !isObject(reader) && isReadAt(reader) {
```

Comment:
There is a possibility by implementing io.SectionReader() based on io.ReadSeeker instead of io.ReaderAt.  But it is not worth the complexity for now in my opinion.